src/react/hooks.js

```diff
@@ -48,3 +52,298 @@ export const useTrackColorSchemeChoice = () => {
+      } else if (themeVariantLink.rel !== stylesheetRelForVariant) {
+        themeVariantLink.rel = stylesheetRelForVariant;
+      }
+    });
+  }, [themeVariants, currentThemeVariant, onLoadThemeVariantLight]);
+};
+
+const handleParagonVersionSubstitution = (url) => {
+  if (!url || !url.includes('$paragonVersion') || !PARAGON?.version) {
+    return url;
+  }
+  return url.replace('$paragonVersion', PARAGON.version);
+};
+
+/**
+ * TODO
+ * @param {*} config
+ * @returns An object containing the URLs for the theme's core CSS and any theme variants.
+ */
+const getParagonThemeUrls = (config) => {
+  const paragonThemeUrls = config.PARAGON_THEME_URLS || {};
+  const coreCssUrl = handleParagonVersionSubstitution(paragonThemeUrls.core);
+
+  const themeVariantsCss = {};
+  Object.entries(paragonThemeUrls.variants || {}).forEach(([themeVariant, { url, ...rest }]) => {
+    themeVariantsCss[themeVariant] = {
+      url: handleParagonVersionSubstitution(url),
+      ...rest,
+    };
+  });
+
+  const hasMissingCssUrls = !coreCssUrl || Object.keys(themeVariantsCss).length === 0;
+  if (hasMissingCssUrls) {
+    if (!PARAGON) {
+      return {};
+    }
+    const themeVariants = {};
+    const prependBaseUrl = (url) => `${config.BASE_URL}/${url}`;
+    Object.entries(PARAGON.themeUrls.variants).forEach(([themeVariant, { fileName, ...rest }]) => {
+      themeVariants[themeVariant] = {
+        url: prependBaseUrl(fileName),
+        ...rest,
+      };
+    });
+    return {
+      core: {
+        url: prependBaseUrl(PARAGON.themeUrls.core),
+      },
+      variants: themeVariants,
+    };
+  }
+
+  return {
+    core: {
+      url: handleParagonVersionSubstitution(coreCssUrl),
+    },
+    variants: themeVariantsCss,
+  };
+};
+
+/**
+ * Given the inputs of URLs to the CSS for the core application theme and the theme variants (e.g., light), this hook
+ * will inject the CSS as `<link>` elements into the HTML document, loading each theme variant's CSS with an appropriate
+ * priority based on whether its the currently active theme variant. This is done using "alternate" stylesheets. That
+ * is,the browser will still download the CSS for the non-current theme variants, but at a lower priority than the
+ * current theme variant's CSS. This ensures that if the theme variant is changed at runtime, the CSS for the new theme
+ * variant will already be loaded.
+ *
+ * @memberof module:React
+ * @param {object} config An object containing the URLs for the theme's core CSS and any theme (i.e., `getConfig()`)
+ *
+ * @returns An array containing 2 elements: 1) an object containing the app
+ *  theme state, and 2) a dispatch function to mutate the app theme state.
+ */
+export const useParagonTheme = (config) => {
+  const paragonThemeUrls = getParagonThemeUrls(config);
+  const {
+    core: themeCore,
+    variants: themeVariants,
+  } = paragonThemeUrls;
+
+  const initialParagonThemeState = {
+    isThemeLoaded: false,
+    themeVariant: 'light',
```

Comment:
TODO: be smart about the default `themeVariant`. Use the `default` property from `paragonThemeUrls`.